rkyv/src/niche/option_box.rs

```diff
@@ -20,6 +20,54 @@ pub struct ArchivedOptionBox<T: ArchivePointee + ?Sized> {
     inner: ArchivedBox<T>,
 }
 
+#[cfg(feature = "bytecheck")]
+const _: () = {
+    use crate::{
+        bytecheck::{CheckBytes, Verify},
+        rancor::Error,
+        validation::{ArchiveContext, LayoutRaw},
+        RelPtr,
+    };
+
+    unsafe impl<T, C> CheckBytes<C> for ArchivedOptionBox<T>
+    where
+        T: ArchivePointee + ?Sized,
+        C: Fallible + ?Sized,
+        ArchivedOptionBox<T>: Verify<C>,
+        RelPtr<T>: CheckBytes<C>,
+    {
+        unsafe fn check_bytes(
+            value: *const Self,
+            context: &mut C,
+        ) -> Result<(), C::Error> {
+            // bypass `ArchivedBox::check_bytes` in favor of `RelPtr::check_bytes`.
+            // both ArchivedOptionBox and ArchivedBox are transparent to the RelPtr,
+            // so casting to RelPtr is safe.
+            RelPtr::check_bytes(value.cast(), context)?;
+
+            // verify with null check
+            Verify::verify(unsafe { &*value }, context)
+        }
+    }
+
+    unsafe impl<T, C> Verify<C> for ArchivedOptionBox<T>
+    where
+        T: ArchivePointee + CheckBytes<C> + LayoutRaw + ?Sized,
+        T::ArchivedMetadata: CheckBytes<C>,
+        C: Fallible + ArchiveContext + ?Sized,
+        C::Error: Error,
+    {
+        #[inline]
+        fn verify(&self, context: &mut C) -> Result<(), C::Error> {
+            if self.inner.is_null() {
+                return Ok(());
```

Comment:
Nit: I would prefer `if .. { Ok(()) } else { self.inner.verify(context) }` to avoid the explicit `return`.